lib/pavilion/test_run.py

```diff
@@ -1090,6 +1090,33 @@ def _write_script(self, stype, path, config):
             script.comment('Output the environment for posterity')
             script.command("declare -p")
 
+        spack_commands = config.get('spack', {})
+        install_packages = spack_commands.get('install', [])
+        load_packages = spack_commands.get('load', [])
+        setup_spack_env = self.config.get('setup_spack_env', 'False')
+        spack_config = self.config.get('spack_config', {})
+
+        if setup_spack_env in ['True', 'true']:
+            script.newline()
+            script.comment('Add Spack path to path.')
+            script.command('export SPACK_ROOT={}'
+                           .format(self._pav_cfg.get('spack_path')))
+            script.command('export PATH=$SPACK_ROOT/bin:$PATH')
+            script.comment('Activate spack environment.')
+            script.command("spack env activate -d .")
+
+        if install_packages:
+            script.newline()
+            script.comment('Install spack packages.')
+            for package in install_packages:
+                script.command('spack install {}'.format(package))
+
+        if load_packages:
+            script.newline()
+            script.comment('Load spack packages.')
+            for package in load_packages:
+                script.command('spack load {}'.format(package))
+
```

Comment:
I'm really glad I looked over your tests, because there are some problems here.

1. All the spack stuff should be conditioned on the `pav_cfg.spack_path` being set. 
2. It should be an error to try and run/build a test with spack enabled without a spack path set.
3. Feel free to raise TestConfigError as appropriate.